src/ssl.c

```diff
@@ -5641,12 +5641,46 @@ int AddCA(WOLFSSL_CERT_MANAGER* cm, DerBuffer** pDer, int type, int verify)
 static int wolfSSL_RAND_InitMutex(void);
 #endif
 
+/* If we don't have static mutex initializers, but we do have static atomic
+ * initializers, activate WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS to leverage
+ * the latter.
+ */
+#ifndef WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS
+    #if !defined(WOLFSSL_MUTEX_INITIALIZER) && !defined(SINGLE_THREADED) && \
+            defined(WOLFSSL_ATOMIC_OPS) && defined(WOLFSSL_ATOMIC_INITIALIZER)
+        #define WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS 1
+    #else
+        #define WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS 0
+    #endif
+#elif defined(WOLFSSL_MUTEX_INITIALIZER) || defined(SINGLE_THREADED)
+    #undef WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS
+    #define WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS 0
+#endif
```

Comment:
Why `undef & define` and just `define` in the other?

Reply:
to force the feature off we define `WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS` to `0`.  the user can also force it off by defining to `0`.

the top half is followed if `WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS` has no definition either way, in which case the default setup is used, following the config settings.

the bottom half is followed if the user config activated `WOLFSSL_CLEANUP_THREADSAFE_BY_ATOMIC_OPS`, but it needs to be forced off because it's not compatible with other config settings.